test/forge/FeeTest.sol

```diff
@@ -348,4 +349,63 @@ contract FeeTest is IntegrationTest {
         assertEq(assets, expectedAssets, "assets");
         assertGe(assets, assetsBefore, "assets increased");
     }
+
+    function testUpdateWithdrawQueueRemovingDisabledMarketWithFee(
+        uint256 firstAmountSupplied,
+        uint256 secondAmountSupplied,
+        address newFeeRecipient,
+        uint256 blocks
+    ) public {
+        firstAmountSupplied = bound(firstAmountSupplied, MIN_TEST_ASSETS, MAX_TEST_ASSETS);
+        secondAmountSupplied = bound(secondAmountSupplied, MIN_TEST_ASSETS, MAX_TEST_ASSETS);
+        blocks = _boundBlocks(blocks);
+        vm.assume(newFeeRecipient != FEE_RECIPIENT && newFeeRecipient != address(0));
+
+        _setCap(allMarkets[0], firstAmountSupplied);
+        _setCap(allMarkets[1], secondAmountSupplied);
+
+        Id[] memory supplyQueue = new Id[](2);
+        supplyQueue[0] = allMarkets[0].id();
+        supplyQueue[1] = allMarkets[1].id();
+
+        vm.prank(ALLOCATOR);
+        vault.setSupplyQueue(supplyQueue);
+
+        loanToken.setBalance(SUPPLIER, firstAmountSupplied + secondAmountSupplied);
+
+        vm.prank(SUPPLIER);
+        vault.deposit(firstAmountSupplied + secondAmountSupplied, ONBEHALF);
+
+        _setCap(allMarkets[1], 0);
+
+        vm.prank(CURATOR);
+        vault.submitMarketRemoval(allMarkets[1].id());
+
+        vm.warp(block.timestamp + TIMELOCK);
+
+        uint256[] memory indexes = new uint256[](2);
+        indexes[0] = 0;
+        indexes[1] = 1;
+
+        vm.prank(ALLOCATOR);
+        vault.updateWithdrawQueue(indexes);
+
+        assertApproxEqAbs(vault.lastTotalAssets(), firstAmountSupplied, 1, "lastTotalAssets");
+
+        _forward(blocks);
```

Comment:
how does it generate interest if there's no borrower?

Reply:
alright we don't care

Reply:
There are borrowers